### PR TITLE
Add ThreadNames test to LinuxTracingIntegrationTest

### PR DESCRIPTION
**Pull request by dpallotti**
Bug: http://b/160756664

### CODE REVIEW

File: src/LinuxTracing/LinuxTracingIntegrationTestPuppet.cpp [hunk 1]
```diff
@@ -60,6 +65,8 @@ int LinuxTracingIntegrationTestPuppetMain() {
     LOG("Puppet received command: %s", command);
     if (command == PuppetConstants::kSleepCommand) {
       SleepRepeatedly();
+    } else if (command == PuppetConstants::kSetnameCommand) {
+      ChangeName();
```
Comment (dimitry-) [inline src/LinuxTracing/LinuxTracingIntegrationTestPuppet.cpp:69]: ChangeCurrentThreadName?

File: src/LinuxTracing/LinuxTracingIntegrationTestPuppet.h [hunk 2]
```diff
@@ -13,10 +13,12 @@ struct LinuxTracingIntegrationTestPuppetConstants {
   LinuxTracingIntegrationTestPuppetConstants() = delete;
 
   constexpr static uint64_t kSleepCount = 1000;
+  constexpr static const char* kNewName = "New Name";
   constexpr static const char* kSharedObjectFileName =
       "libLinuxTracingIntegrationTestPuppetSharedObject.so";
 
   constexpr static const char* kSleepCommand = "sleep";
+  constexpr static const char* kSetnameCommand = "setname";
```
Comment (dimitry-) [inline src/LinuxTracing/LinuxTracingIntegrationTestPuppet.h:21]: also maybe this set(main)threadname

**Review by dimitry- — APPROVED** [review]
otherwise lgtm

**Review by florian-kuebler — APPROVED** [review]
Despite the clang-format issue, LGTM!

File: src/LinuxTracing/LinuxTracingIntegrationTestPuppet.h (hunk 2 above)
Comment (dpallotti) [reply src/LinuxTracing/LinuxTracingIntegrationTestPuppet.h:21]: The variable or the string? The idea is to kind of match the calls they make: `pthread_setname_np` -> `setname`.

File: src/LinuxTracing/LinuxTracingIntegrationTestPuppet.cpp (hunk 1 above)
Comment (dpallotti) [reply src/LinuxTracing/LinuxTracingIntegrationTestPuppet.cpp:69]: `ChangeMainThreadName`?

Comment (dpallotti) [reply src/LinuxTracing/LinuxTracingIntegrationTestPuppet.cpp:69]: Well ok, it can be called from anywhere, I'll change to `ChangeCurrentThreadName`.

Comment (dpallotti) [reply src/LinuxTracing/LinuxTracingIntegrationTestPuppet.cpp:69]: Done.

File: src/LinuxTracing/LinuxTracingIntegrationTestPuppet.h (hunk 2 above)
Comment (dimitry-) [reply src/LinuxTracing/LinuxTracingIntegrationTestPuppet.h:21]: maybe then align value with the name of libc function? so pthread_setname_np in this case? setname seems too generic.

Comment (dpallotti) [reply src/LinuxTracing/LinuxTracingIntegrationTestPuppet.h:21]: Done.

**Pull request merged**